### DISCUSSION TITLE
Should I declare Hadamard gate before calling the Basis Encoding method

### DISCUSSION THREAD

**Discussion by HuangChiEn** [PennyLaneAI/pennylane#1613] [Q&A]
While I saw the source code of Basis encoding, I just wonder that should I declare Hadamard gate before calling the Basis Encoding method ? Since the Basis Encoding just wraps the non-zero bit with the PauliX gate (bit flip), while it does not work without putting the initial state into superposition. 

![image](https://user-images.githubusercontent.com/52521165/131466054-b18e568e-a49f-4fdf-9f49-b4be244cc922.png)

By the way, I also have no idea how to put the suitable place (wires) of Hadamard gate, the number of Hadamard gate depends on the number of data you have.
However, the measurement result will be various according to the place you set the Hadamard gate.

For example, see the pennylane tutorial : 
[tutorial](https://pennylane.ai/qml/glossary/quantum_embedding.html)
I solve it, and I should place the H gate on the q1 wires, but I would not solve it every time manually, especially in the multi-qubit (e.g. num_qubit=16) !!

![image](https://user-images.githubusercontent.com/52521165/131466253-9392e17a-0633-43a6-93f5-4df65a46c441.png)

Any suggestion will be appreciated !!

**Comment by josh146** [comment]
Hi @HuangChiEn! I believe this is working as expected, no Hadamard is needed 🙂 

With the basis embedding, we are associating each input with a computational basis of the qubit system. For example, consider input data `x = [1, 0, 0, 1]`. We can embed this into a 4-qubit system by flipping the state of the first and last qubits using PauliX gates:

```python
PauliX(0)
PauliX(3)
```

For more details, check out the [PennyLane glossary](https://pennylane.ai/qml/glossary/quantum_embedding.html#basis-embedding) :)

**Comment by josh146** [comment]
@HuangChiEn hope this helps! If you have any other questions, please feel free to open a post over at the [PennyLane discussion forum](https://discuss.pennylane.ai/). The forum is actively moderated and visited by PennyLane developers, unlike the GitHub discussion forum.
